plugins/mimecast/komand_mimecast/tasks/monitor_siem_logs/task.py

```diff
@@ -59,11 +72,12 @@ def run(  # pylint: disable=unused-argument # noqa: MC0001
                         f"Error: {error}, returning state={state}, has_more_pages={has_more_pages}"
                     )
                     return [], state, has_more_pages, error.status_code, error
-                output = self._filter_and_sort_recent_events(output, filter_time)
+
+                output, last_log_line = self._filter_and_sort_recent_events(output, filter_time, last_log_line)
```

Comment:
I do think we need to perform sort of hash to check the next header token is still returning the same datset (even for first run when we have no token, although for these we maybe would filter all out but in a backfill may need to access the same file?) otherwise we could be skipping indexes. I know the file name changes because it's built dynamically, but is it possible (but horrible) to do a `shasum(json.dumps(output))` otherwise we should drop the last_log_line?

Reply:
yeah I was trying to look at some methods to has the files, i'll explore some options

Reply:
I was looking at `_handle_zip_file` and we are able to grab a list of the name of files, I wonder at this point could we run a hash against this and store it in the state for comparison if we stop mid-processing?